repowiz-domain/src/main/groovy/life/qbic/repowiz/find/FindMatchingRepositories.groovy

```diff
@@ -7,17 +7,32 @@ import life.qbic.repowiz.find.tree.Node
 import org.apache.logging.log4j.LogManager
 import org.apache.logging.log4j.Logger
 
-class FindMatchingRepositories implements FindMatchingRepositoriesInput{
+/**
+ * This class is responsible for finding a matching repository for a users data
+ *
+ * This class should be used whenever a matching repository needs to be found. A decision tree is used in order to find a repository.
+ *
+ *  @since: 1.0.0
+ *  @author: Jennifer Bödker
+ *
+ */
```

Comment:
```suggestion
/**
 * This class is responsible for finding a matching repository for data that was submitted by the user.
 *
 * This class should be used whenever a matching repository needs to be found. A decision tree is used in order to find a repository.
 *
 *  @since: 1.0.0
 *  @author: Jennifer Bödker
 *
 */
```